lib/blockscore/base.rb

```diff
@@ -66,18 +67,20 @@ def capture_attributes(source)
       @attributes = source.attributes
     end
 
-    def add_accessor(symbol, *_args)
+    def add_accessor(symbol)
       singleton_class.instance_eval do
         define_method(symbol) do
-          wrap_attribute(attributes[symbol])
+          wrap_attribute(attributes.fetch(symbol))
         end
       end
     end
 
-    def add_setter(symbol, *_args)
+    def add_setter(symbol)
+      attr = symbol.to_s.chop.to_sym
+      assert_mutable(attr)
       singleton_class.instance_eval do
```

Comment:
Yes and no. Presently it is designed such that anything that is a valid method name can be used as an attribute. The backend decides whether it is going to silently discard it. And AFAICT that is desired behavior. Personally I would want to know about typos if I was a developer. We could probably define the extent of the values, but moving that to the API code means it has to be changed everytime the back end changes. Not saying that is often, but they need to obviously stay in sync. So far when I have floated that concept it seems like that is reserved for next version of API.